lib/sqewer/local_connection.rb

```diff
@@ -0,0 +1,134 @@
+class Sqewer::LocalConnection < Sqewer::Connection
+  FAIL_AFTER_DELIVERIES = 10
+
+  def with_db(**k)
+    SQLite3::Database.open('sqewer-local-queue.sqlite3', **k) do |db|
+      db.busy_timeout = 5
+      return yield db
+    end
+  end
+
+  def with_readonly_db(&blk)
+    with_db(readonly: true, &blk)
+  end
+
+  def initialize(queue_url)
+    require 'sqlite3'
```

Comment:
Should this require be at the top of the file for style purposes?

Reply:
No, because it's a "peer dependency" - if we require it at the top we force every system using sqewer to also install sqlite3 just so that this file could be loaded.